docs/docs/features/keymaps.mdx

```diff
@@ -38,21 +37,18 @@ For the full set of possible behaviors, see the [overview page for behaviors](..
 ## Layers
 
 Like many mechanical keyboard firmwares, ZMK keymaps are composed of a collection of layers, with a
-minimum of at least one layer that is the default, usually near the bottom of the "stack". Each layer
+minimum of at least one layer that is the default, usually near the bottom of the "layer stack". Each layer
 in ZMK contains a set of bindings that bind a certain behavior to a certain key position in that layer.
 
 |                                                   ![Diagram of three layers](../assets/features/keymaps/layer-diagram.png)                                                    |
 | :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
 | _A simplified diagram showing three layers. The layout of each layer is the same (they all contain four keys), but the behavior bindings within each layer can be different._ |
 
-In addition to the base default layer (which can be changed), certain bound behaviors may also
-enable/disable additional layers "on top" of the default layer.
+All layers are assigned and referred to by a natural number, with the base layer being layer `0`. It is common to [use the C preprocessor to "name" layers](../behaviors/layers.md#defines-to-refer-to-layers), making them more legible.
 
-When a key location is pressed/released, the stack of all active layers from "top to bottom" is used,
-and the event is sent to the behavior bound at that position in each layer, for it to perform whatever
-actions it wants to in reaction to the event. Those behaviors can choose to "handle" the event, and stop
-it from being passed to any lower layers, or may choose to "pass it along", and let the next layer
-in the stack _also_ get the event.
+The default layer (usually the base layer) is always enabled. Certain bound behaviors may enable/disable additional layers.
+
+When a key location is pressed/released, the _highest-valued currently active_ layer is used. The press/release event is sent to the behavior bound at that position in said layer, for it to perform whatever actions it wants to in reaction to the event. The behavior can choose to "consume" the event, or "pass it along" and let the next highest-valued active layer _also_ get the event (whose behavior may continue "passing it along").
 
```

Comment:
I think it would be worth adding a "tip" box here, talking about multiple base layouts. That seems to be the most common scenario where people have issues.

i.e. if you have toggled layers like qwerty vs. colemak, you should always place them first in the stack (defined earlier) so that other layers that can be activated (momentarily) through them work as expected.

This might be worth a more elaborate section such as the one mentioned in #1953, but for now a short tip block would still be helpful. If we add that, then we can simply link there from here.

Reply:
I agree on both points, added tip block for now. When #1953 is done we can revisit.